src/dbus_fast/aio/proxy_object.py

```diff
@@ -103,16 +103,22 @@ async def method_fn(*args, flags=MessageFlag.NONE):
 
             if not out_len:
                 return None
-            elif out_len == 1:
+
+            if flags & MessageFlag.UNPACK_VARIANTS:
+                body = unpack_variants(body)
+
+            if out_len == 1:
                 return body[0]
-            else:
-                return body
+            return body
 
         method_name = f"call_{BaseProxyInterface._to_snake_case(intr_method.name)}"
         setattr(self, method_name, method_fn)
 
-    def _add_property(self, intr_property):
-        async def property_getter():
+    def _add_property(
+        self,
+        intr_property,
+    ):
+        async def property_getter(*, flags=MessageFlag.NONE):
             msg = await self.bus.call(
```

Comment:
In a future PR it would be good to move the unpack variants as part of `call` as well so we don't create them in the first place

Reply:
Ok yea that makes sense. I wasn't sure if we wanted it in the low-level client but I knew even if we did the high-level client needed the variants returned for now because otherwise this bit would fail:
https://github.com/Bluetooth-Devices/dbus-fast/blob/f4d173e0426990cd0b8b1b813949cd5027684102/src/dbus_fast/aio/proxy_object.py#L98-L103

We'd have to refactor that out as well if the high level client is not being returned messages that contain variants.

Reply:
It is a bit difficult since everything is blindly unpacked.  Anyways, something for the future if we can figure out a workable solution